keys-controller.js

```diff
@@ -88,6 +95,15 @@ function factory($scope, $routeParams, brAlertService, brKeyService,
   });
 }
 
+// TODO: allow users to modify specific resources
+function hasPermission(identity, permission) {
+  // if sysPermissionTable is an object, it is a map of specific resources
+  // to which this permission applies.
+  // boolean value indicates that the user has unrestricted permissions
+  return (permission in identity.sysPermissionTable &&
+    typeof identity.sysPermissionTable[permission] === 'boolean');
```

Comment:
sysPermission could be a boolean or an object.  So a simple if(sysPermissionTable[permission] === true) will resolve true if sysPermission is an object which is not desirable.  In the 'admin' case, the boolean will always = true.